### DISCUSSION TITLE
replacing-disqus-with-github-discussions-using-giscus/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#2] [Announcements]
# replacing-disqus-with-github-discussions-using-giscus/

I recently replaced Disqus with giscus for comments on this blog. In this post I describe the process I took

https://andrewlock.net/replacing-disqus-with-github-discussions-using-giscus/

**Comment by SeanKilleen** [comment]
Congratulations! 🎉 Thanks for the link. This is a good reminder to me that I should draft these sorts of posts as I do things. I have a backlog item called "write up how you moved to giscus". 😅

**Reply by andrewlock** [reply]
Thanks! 😃 And thanks for providing the links - it was great to have your changes to refer to when I was thinking, "is this _really_ all there is to it?!" 😅

**Comment by thomaslevesque** [comment]
Congrats for making the switch!

Just one remark: the theme for the comments area is set to dark, which looks strange when viewing your blog in light theme (I just realized you had a theme switch!). Ideally, it should follow your blog's theme, not the user's preferred Github theme.

**Reply by shoaibshakeel381** [reply]
can you tell me where's the them switch. It's really hard to read comments with high contrast of black and white.

**Reply by shoaibshakeel381** [reply]
or are you talking about github's global theme switch maybe?

**Reply by thomaslevesque** [reply]
@shoaibshakeel381 actually it doesn't seem to follow the user's preferred Github theme. I think it's just always dark.

**Reply by shoaibshakeel381** [reply]
yeah, I guess andrew just set preferred color scheme to amoled dark.

**Comment by dimohy** [comment]
Congratulations! 🎉🎉🎉

I'm on `GitHub`, not on `Disqus`, so I'm free to comment!

**Comment by KoalaBear84** [comment]
Great, also familiar environment when you are already using GitHub. **Nice** 👍

_Edit also works when you are going to GitHub._

**Comment by skovsende** [comment]
I tried replying on the last post about how you should move - but Discuss was being a hassle, so I didn't. So I'm very happy you moved away! :)

**Comment by fabsenet** [comment]
Great! Just wanted to write something to test this :)

**Comment by kaylumah** [comment]
Your first post about the switch coincided with a friend mentioning a new comment platform, I was using a homegrown version using the GitHub Issue API (which does not support discussions) so this is much better.

Thank you for the addition regarding `giscuss.json` added that now as wel

**Comment by skolima** [comment]
Nice one - I've just setup a new blog using GitHub+Jekyll, so those GitHub-hosted comments (and the engine to run them) are an especially nice find.

**Comment by VaclavElias** [comment]
I am planning to do the same for open source project [Stride](https://www.stride3d.net/). We are currently using there Disqus. How do you handle spam?

**Reply by andrewlock** [reply]
I've only recently switched but haven't seen any spam yet. I guess the GitHub login requirement is a good filter 🤷‍♂️

**Comment by francofgp** [comment]
Nice I was looking for alternative to disqus, I will try to implement this on my own blog. It looks awesome.

Just one comment related to your blog. I would prefer a light theme for your blog.

**Reply by francofgp** [reply]
Me again, It literally took me less than 10 minutes to add Giscus to my blog, thanks!!!

**Comment by markheath** [comment]
Thanks for this post. I'm going through the same process now, but have got stuck trying to auto-import my old comments as GitHub discussions. Seems it's [not supported at all in Octokit](https://github.com/octokit/octokit.net/issues/2254) and [possibly not in Octokit GraphQL](https://github.com/octokit/octokit.graphql.net/issues/253) either (although there does seem to be some Discussion-related code in the repo)

**Comment by bric3** [comment]
Hey thank you for this article. It helped me achieve the migration as well.
Regarding the migration I wrote a small program to do that, in Java. I won;t claim it's bug free, but it handles quite a few things. If that helps others.

https://blog.arkey.fr/2022/10/16/moving-from-disqus-to-giscus/

**Comment by kongh** [comment]
Congratulations!

**Comment by caseyspaulding** [comment]
Yea this is so much better . I got to try it.

**Comment by anittasaju1996** [comment]
Great blog! Thanks

**Comment by mattpopovich** [comment]
There are a few things that I think will push me to make the switch. No ads is great (although I didn't see any Disqus ads in my comment section yet), but there were a few things that annoyed me on my software-related blog:
* URLs people commented were changed to route through disqus, making it difficult to tell where the link was actually going. Ex. google.com was changed to https://disq.us/url?url=https%3A%2F%2Fgoogle.com%3Aasdlkfj&cuid=sdflkjsadf
* Links commented in Disqus would be auto shortened with [...]. You could only see the full path when hovering over it. 
* Link / code / all formatting [was done via HTML](https://disqus.com/home/discussion/channel-discussdisqus/markdown_support/). I think Markdown is the current standard (Stack Overflow, GitHub, etc.). I think it will be a welcome change. 

Thanks for your blog post. I think I'm going to work on making the switch this weekend. 

Cheers!

**Comment by MortezaMirmojarabian** [comment]
It seems to be great! Thanks for the article and simple instructions. I have a question, please: Do I need a separate repository for each of my blog posts?
